nekoyume/Assets/_Scripts/State/States.cs

```diff
@@ -157,27 +153,34 @@ private static async UniTask<AvatarState> GetAvatarStateAsync(Address address, b
                 return new AvatarState(dict);
             }
 
-            foreach (var key in new[]
+            var addressPairList = new List<string>
             {
                 LegacyInventoryKey,
                 LegacyWorldInformationKey,
-                LegacyQuestListKey,
-            })
+                LegacyQuestListKey
+            }.Select(key => (Key: key, KeyAddress: address.Derive(key))).ToArray();
+
+            var states = await agent.GetStateBulk(addressPairList.Select(value => value.KeyAddress));
+            // Make Tuple list by state value and state address key.
+            var stateAndKeys = states
+                .Join(addressPairList,
+                    state => state.Key,
+                    addressKeyTuple => addressKeyTuple.KeyAddress,
```

Comment:
It is better to use `addressPairList` with `addressPair` or use `addressKeyTuples` with `addressKeyTuple`.

Reply:
I reflected it on [cbbce8c](https://github.com/planetarium/NineChronicles/pull/1158/commits/cbbce8cd024a5ffbb89ec16c7ba6a1a283db41b2) commit!